emu_base/pulser_adapter.py

```diff
@@ -117,34 +120,63 @@ def _extract_omega_delta_phi(
     nsamples = len(target_times) - 1
     omega = torch.zeros(
         nsamples,
-        len(sequence.register.qubit_ids),
+        len(q_ids),
         dtype=torch.complex128,
     )
 
     delta = torch.zeros(
         nsamples,
-        len(sequence.register.qubit_ids),
+        len(q_ids),
         dtype=torch.complex128,
     )
     phi = torch.zeros(
         nsamples,
-        len(sequence.register.qubit_ids),
+        len(q_ids),
         dtype=torch.complex128,
     )
+    qubit_positions = _get_qubit_positions(sequence.register)
 
-    if laser_waist:
-        qubit_positions = _get_qubit_positions(sequence.register)
-        waist_factors = torch.tensor(
-            [math.exp(-((x[:2].norm() / laser_waist) ** 2)) for x in qubit_positions]
-        )
-    else:
-        waist_factors = torch.ones(len(sequence.register.qubit_ids))
+    def perp_dist(pos: torch.Tensor, axis: torch.Tensor) -> Any:
```

Comment:
why you return `Any` ? it seems that `torch.linalg.vector_norm` returns `Tensor`

Reply:
it would seem so at first sight, but it actually returns `Any`... Maybe it's an issue with my mypy environment, but it's like that for me.